### DISCUSSION TITLE
How to flash?

### DISCUSSION THREAD

**Discussion by AlexGuld** [UtilitechAS/amsreader-firmware#710] [General]
Hi, I'm having som trouble flashing a NodeMCU ESP32.

according to the wiki I need 3 files. I can only find 1.

the guide says I need bootloader.bin, partitions.bin and firmware.bin.

In the release section there is only one file per device. I'm guessing that is firmware. so where do I get bootloader and partitions?
the link for bootloader is not working.

**Comment by AlexGuld** [comment]
Newer mind, it's all included in the zip version.

**Comment by AlexGuld** [comment]
Perhaps the wiki needs an update?

**Reply by ArnieO** [reply]
You're right, that was outdated. Thank you for reminder.
Better now? https://github.com/UtilitechAS/amsreader-firmware/wiki/Flashing

**Comment by Noschvie** [comment]
@ArnieO is the usage and support of "ESP Web Tools" planned? Thanks!

**Reply by ArnieO** [reply]
No, and I am personally not aware of the potential benefits. Maybe @gskjold is familiar with ESP Web Tools. 
Please feel free to argue for it! 😃
